runtime/src/trace.rs

```diff
@@ -0,0 +1,498 @@
+            fn record_i128(&mut self, field: &Field, value: i128) {
+                match i64::try_from(value) {
+                    Ok(value) => self.record_i64(field, value),
+                    Err(_) => <Self as SpanFieldVisitor>::visit(
+                        self,
+                        field.name(),
+                        value.to_string().into(),
+                    ),
+                }
+            }
+
+            #[inline(always)]
+            fn record_u128(&mut self, field: &Field, value: u128) {
+                match i64::try_from(value) {
+                    Ok(value) => self.record_i64(field, value),
+                    Err(_) => <Self as SpanFieldVisitor>::visit(
+                        self,
+                        field.name(),
+                        value.to_string().into(),
+                    ),
+                }
+            }
+
+            #[inline(always)]
+            fn record_bool(&mut self, field: &Field, value: bool) {
+                <Self as SpanFieldVisitor>::visit(self, field.name(), value.into());
+            }
+
+            #[inline(always)]
+            fn record_str(&mut self, field: &Field, value: &str) {
+                <Self as SpanFieldVisitor>::visit(self, field.name(), value.to_string().into());
+            }
+
+            #[inline(always)]
+            fn record_bytes(&mut self, field: &Field, value: &[u8]) {
+                <Self as SpanFieldVisitor>::visit(
+                    self,
+                    field.name(),
+                    opentelemetry::logs::AnyValue::Bytes(Box::new(value.to_vec())),
+                );
+            }
+
+            #[inline(always)]
+            fn record_error(&mut self, field: &Field, value: &(dyn 'static + std::error::Error)) {
+                <Self as SpanFieldVisitor>::visit(self, field.name(), value.to_string().into());
+            }
+
+            #[inline(always)]
+            fn record_debug(&mut self, field: &Field, value: &dyn std::fmt::Debug) {
+                <Self as SpanFieldVisitor>::visit(
+                    self,
+                    field.name(),
+                    format!("{:?}", value).into(),
+                );
+            }
+        }
+    };
+}
+
+#[derive(Clone, Debug, Default)]
+struct SpanFieldValues(BTreeMap<opentelemetry::Key, opentelemetry::logs::AnyValue>);
+
+impl SpanFieldVisitor for SpanFieldValues {
+    fn visit(&mut self, key: &'static str, value: opentelemetry::logs::AnyValue) {
+        self.0.insert(key.into(), value);
+    }
+}
+
+impl_visit!(SpanFieldValues);
+
+#[derive(Clone, Debug, Default)]
+struct EventFieldValues {
+    data: SpanFieldValues,
+    message: Option<opentelemetry::logs::AnyValue>,
+}
+
+impl EventFieldValues {
+    const REMAPPED_FIELDS: [(&'static str, &'static str); 4] = [
+        ("log.line", CODE_LINENO),
+        ("log.file", CODE_FILEPATH),
+        ("log.target", "code.target"),
+        ("log.module_path", "code.module_path"),
+    ];
+}
```

Comment:
I see we have a lot of code to remap these fields to match otel conventions. Do you mind expanding on why we need to do that, and what would happen if we didn't? Is it a requirement, or a nice to have?

Reply:
It's... sort of both? It's less "remapped to match convention" and more "standardized so it all comes out the same". If we *don't* do it, the queries on the user side get really annoying because the attributes show up differently based on whether an event was emitted by `log` or `tracing`. The convention is just a convenient target to standardize on 😅

Reply:
Okay, sgtm, thanks!